### ISSUE TITLE
Configuration interface incorrect when setting withDefaultLockAtLeastFor

### ISSUE THREAD

**Issue by stejanse** [lukas-krecan/ShedLock#32]
When I create a `ScheduledLockConfiguration` and try to set `withDefaultLockAtLeastFor`, I cannot build the configuration anymore without casting. The following example does not compile:

```
    @Bean
    public ScheduledLockConfiguration taskScheduler(LockProvider lockProvider) {
        return ScheduledLockConfigurationBuilder
                .withLockProvider(lockProvider)
                .withPoolSize(10)
                .withDefaultLockAtMostFor(Duration.ofMinutes(15))
                .withDefaultLockAtLeastFor(Duration.ofSeconds(5))
                .build();
    }
```

The workaround is casting the `ConfigurationBuilder` after calling `withDefaultLockAtLeastFor`:

```
    @Bean
    public ScheduledLockConfiguration taskScheduler(LockProvider lockProvider) {
        ScheduledLockConfigurationBuilder.ConfiguredScheduledLockConfigurationBuilder builder = 
                (ScheduledLockConfigurationBuilder.ConfiguredScheduledLockConfigurationBuilder) ScheduledLockConfigurationBuilder
                .withLockProvider(lockProvider)
                .withPoolSize(10)
                .withDefaultLockAtMostFor(Duration.ofMinutes(15))
                .withDefaultLockAtLeastFor(Duration.ofSeconds(5));
        return builder.build();
    }
```

In my opinion, the subinterfaces should be removed from the `ScheduledLockConfigurationBuilder` interface.

**Comment by lukas-krecan** [comment]
Hi, thanks for feedback. Will fix it in the next release.

**Comment by lukas-krecan** [comment]
Released as 0.15.0
